app/main.py

```diff
@@ -1,13 +1,25 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error=error) for error in errors],
+        "path": file_path,
+        "status": "passed" if all(
+            [True if len(error) == 0 else False for error in errors]
+        ) else "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(file_path=key, errors=report)
+        for key, report in linter_report.items()
```

Comment:
key is not descriptive variable name

Reply:
This task is still not completed successfully.  What I need to fix? TY